File: packages/notifications/src/telegram.ts
import axios from 'axios';

export class TelegramNotifier {
  private botToken: string;
  private chatIds: string[]; 
  private telegramApiUrl: string;

  constructor() {
    this.botToken = process.env.TELEGRAM_BOT_TOKEN || '';
    
    const rawChatIds = process.env.TELEGRAM_CHAT_ID || '-1003491841685,-1003793881886,-1003907426334';
    
    this.chatIds = rawChatIds.split(',').map(id => id.trim()).filter(Boolean);
    
    this.telegramApiUrl = `https://api.telegram.org/bot${this.botToken}/sendMessage`;
  }

  /**
   * إرسال تنبيه فوري مبثوق لجميع القنوات (Pro, Elite, SaaS)
   */
  public async sendAlert(message: string, severity: 'INFO' | 'WARNING' | 'CRITICAL' = 'INFO') {
    if (!this.botToken) {
      console.error('Telegram Notifier ERROR: TELEGRAM_BOT_TOKEN is missing from .env file.');
      return;
    }

    if (this.chatIds.length === 0) {
      console.error('Telegram Notifier ERROR: No Chat IDs provided.');
      return;
    }

    const emoji = severity === 'CRITICAL' ? '🚨' : severity === 'WARNING' ? '⚠️' : 'ℹ️';
    
    const formattedMessage = `
${emoji} *NDSP SYSTEM ALERT* ${emoji}
----------------------------------
*Severity:* ${severity}
*Message:* ${message}
*Timestamp:* ${new Date().toISOString()}
----------------------------------
    `;

    // بث الرسالة لكل قناة في نفس الوقت عبر الـ Loop
    for (const chatId of this.chatIds) {
      try {
        await axios.post(this.telegramApiUrl, {
          chat_id: chatId,
          text: formattedMessage,
          parse_mode: 'Markdown',
        });
        console.log(`[Success] Alert broadcasted to channel: ${chatId}`);
      } catch (error: any) {
        console.error(`[Error] Failed to send alert to channel ${chatId}:`, error.response?.data || error.message);
      }
    }
  }
}

export const telegramNotifier = new TelegramNotifier();

File: backend/app/integrations/telegram/telegram.py
import os
import requests

TOKEN = os.getenv("TELEGRAM_BOT_TOKEN") or "REVOKED_TOKEN_DO_NOT_USE"
CHAT_IDS = os.getenv("TELEGRAM_CHAT_IDS") or "-1003491841685,-1003793881886"

def send_message(text):
    if not CHAT_IDS:
        print("❌ TELEGRAM_CHAT_IDS NOT SET")
        return

    ids = CHAT_IDS.split(",")

    for chat_id in ids:
        try:
            url = f"https://api.telegram.org/bot{TOKEN}/sendMessage"
            r = requests.post(url, json={
                "chat_id": chat_id.strip(),
                "text": text,
                "parse_mode": "Markdown"
            })
            print("Telegram response:", r.text)

        except Exception as e:
            print("Telegram Error:", e)
